central/detection/lifecycle/manager_impl.go

```diff
@@ -170,7 +179,15 @@ func (m *managerImpl) flushIndicatorQueue() {
 
 	// Group the processes into particular baseline segments
 	baselineMap := make(map[processBaselineKey][]*storage.ProcessIndicator)
+
 	for _, indicator := range indicatorSlice {
+		// Do not add it to the baseline map if we are in the observation period for that deployment
+		if features.PostgresDatastore.Enabled() {
```

Comment:
This looks like a generic optimization. We do not have to put it behind the postgres feature flag.